feat(docs): Add core @api-ts/io-ts-http docs to Docusaurus (#1037)
BitGo/api-ts: website/docs/reference/io-ts-http/combinators/flattened.md

# `flattened`

### Overview

> Creates a codec that decodes a nested structure into a flat object and encodes a flat

Reviewer: ```suggestion
The `flattened` combinator creates a codec that decodes a nested structure into a flat
```

It's odd that this sentence omits the subject. Add the implied subject to the sentence so we're not talking around it.

object back into a nested structure.

### Specification

Accepts two arguments:

- `name`: (`string`) A name for the codec, used in error messages.
- `nestedProps`: (`object`) An object that defines the encoded nested structure. Each
  key represents a top-level property in the encoded form, and its value is an object
  that maps the nested keys to their `io-ts` codecs.

Returns a new codec.

### Behavior

- Decoding: Takes the nested input (matching the `nestedProps` structure) and outputs a
  flat object that contains all the inner properties.
- Encoding: Takes the flat object (containing keys from the inner structures) and
  outputs the nested structure defined by `nestedProps`.

### Caveats

- Defining multiple nested properties with the same name across different top-level keys

>   can lead to undefined behavior. The library tries to prevent this statically where
>   possible.

Reviewer: ```suggestion
  can lead to undefined behavior. The library tries to prevent this where statically
```
